homeassistant/components/screenlogic/entity.py

```diff
@@ -55,7 +55,8 @@ def __init__(
         self._data_path = (*self.entity_description.data_root, self._data_key)
         mac = self.mac
         self._attr_unique_id = f"{mac}_{generate_unique_id(*self._data_path)}"
-        self._attr_name = self.entity_data[ATTR.NAME]
+        if not entity_description.translation_key:
+            self._attr_name = self.entity_data[ATTR.NAME]
```

Comment:
What's this name?

Reply:
Currently all entities get their name from the api.  This leaves that convention in place for entities that get their name from the _device_ (user named circuits/features), which as it applies to the integration, are `switch` and `light` entities, and `sensor` entities for the pump(s), which are named by the api after the first circuit associated with them in the device.

Reply:
Can you add some examples with what you want to achieve where? With translations set, you usually don't need to use `_attr_name` anymore, except for when the entity needs to be the device name

Reply:
The api lib [screenlogicpy](https://github.com/dieselrabbit/screenlogicpy/tree/master) (which I maintain) originally provided the majority of data that HA uses, for each value intended to be supported in HA. 
Example of data provided by the api lib:
```
"air_temperature": {
  "name": "Air Temperature",
  "value": 62,
  "unit": "\u00b0F",
  "device_type": "temperature",
  "state_type": "measurement"
},
"ph": {
  "name": "pH",
  "value": 7.5,
  "unit": "pH",
  "state_type": "measurement"
},
"orp": {
  "name": "ORP",
  "value": 726,
  "unit": "mV",
  "state_type": "measurement"
},
```
Since moving to EntityDescriptions though, the majority of that extra data is no longer used, except for the name.  Prior to this PR adding entity translations, all entities were sill using the name provided by the api lib.  That's where the `self._attr_name = self.entity_data[ATTR.NAME]` came from.

Some of the entities this integration exposes (like the example above) are statically named, meaning the integration knows ahead of time what value it is exposing, but there are other entities that are created dynamically.  Because the ScreenLogic gateway device supports various equipment configurations there can be different amounts of circuits, pumps, and heaters (along with other equipment either present or not).  Additionally, the pool controller allows the user to custom name the individual circuits within the device.  It is intended for entities that expose these circuits ([`switch`](https://github.com/home-assistant/core/blob/9711816542b581d7c40a3d69668c3711d5af7c23/homeassistant/components/screenlogic/switch.py#L38-L60), [`light`](https://github.com/home-assistant/core/blob/9711816542b581d7c40a3d69668c3711d5af7c23/homeassistant/components/screenlogic/light.py#L31-L53) to use the name the user has set in the pool controller device.

Therefore for the entities known ahead of time I've provided translations for them, but for the ones not known ahead of time, the integration "falls-back" to the name provided by the api lib, which is what the end user has set within the device itself.

Sorry it's a bit long-winded, but I thought it might be helpful to have a little context as to how the code got to where it is now.

Reply:
Okay this explains it well. I'll check the code on how we can do this the best. How do the devices currently look in HA? Is every pump its own device? Or is everything thrown under 1 device?

Reply:
Everything is currently under one device.  

There's a mix of conventions within the Pentair ScreenLogic system that made breaking out individual devices messy and in my opinion confusing.  Some equipment is presented as a stand-alone device, some are only referenced through the central controller, and some are presented as multiple devices that are only active depending on what mode the central controller is in.

Because the official app presents as one point of connection/device I elected to start there.

Reply:
Ideally we have one device per physical device. I don't think there's a big lack of user experience if we can't do that and just put it on the controller when that's not possible. How do you mean with the device being active?

Using devices is quite a nice way to not use a lot of placeholders. We can use via_device to neatly connect them to the main controller.

(Also, I don't see this as a must, if we can't do that and the user experience is worse, we shouldn't, I'm just exploring at this point :) )

Reply:
I've been thinking about having each physical device represented as an individual device in HA for some time. I've even started rewriting the api lib in a more OO way to both eventually support that, and also to organize my thoughts on functionality and UX.  In the official apps, some functionality is grouped by type (switches, lights, climate controls), yet other functionality is grouped by device (chemistry controller, salt cell).

One of the non-standard use cases is the climate controls. The app (and api) expose separate heater controls for when the system is in pool mode, or spa mode, and both sets of controls are able to be set at any time, but the current temperature only updates in the one for the currently active mode (pool or spa). Currently this creates two climate entities, both of which would be attached to one physical heater device.  Not impossible to do, but doesn't exactly follow the convention of one main entity for a device.

There's other questions in my mind, like "Which is the root device, the main pool controller or the gateway that we're actually connecting to?", but those are more implementation details.

Reply:
> "Which is the root device, the main pool controller or the gateway that we're actually connecting to?"

You can also create manual devices that don't expose entities. They're not that useful since they don't expose entities, but they do show the device topology better.

> convention of one main entity for a device

But that isn't necessarily a convention. If it makes sense for the user, we don't mind. So for example, you could have 2 climate entities for 1 device and just call one `Pool mode` and the other `Spa mode`. Another option you could think of, is creating a switch entity that switches between pool mode and spa mode. It would then change the mode in the device and change the data the entity uses. (But this all depends on the possible features per mode and if that makes sense for the end user)

Reply:
I thought I had read the "main entity for a device" somewhere, but of course I can't find it. I guess I was mixing up different parts of the documentation in my head.